kb-error.js

```diff
@@ -5,6 +5,6 @@ module.exports = class kbError extends Error {
    */
   constructor(message) {
     super(message);
-    this.name = 'kbError';
+    this.name = this.constructor.name;
```

Comment:
https://javascript.info/custom-errors#breadcrumb-2:~:text=Now%20custom%20errors%20are%20much%20shorter%2C,%3D%20...%22%20line%20in%20the%20constructor.

Also explained in commit message

Reply:
The problem with this approach is when things get uglified and you lose the name.
I have no problem with this change as we're currently not uglifying, but we need to keep this in mind.

Reply:
Good Point!

But right now, I think this doesn't support browser as-is since it doesn't support RequireJS (because of the `module.export = ` part).

Even so, What you're worried about is easily fixable imo.

1. we can always just use it as you intended when requiring on client-side, setting the name in the extended class
2. Most minification\uglification libraries have support for this kind of exclusions.
    You can just not minify class names. That might be too project-wide but it's an option :-)
     ```javascript
        module.exports = {
          entry: './kb-error.js',
          output: {
            filename: 'kb-error.min.js',
            path: path.resolve(__dirname, 'dist'),
          },
          optimization: {
            minimize: true,
            minimizer: [ new TerserPlugin({
              terserOptions: {
                mangle: {
                  keep_classnames: true
                }
              }
            }) ],
          }
        };
     ```

But then again, using:
```javascript
class FooError extends kbError {
      constructor(message) {
        super(message);
      }
    }
```
should work without modification (node.js for example)

and:
```javascript
class FooError extends kbError {
      constructor(message) {
        super(message);
        this.name = 'FooError';
      }
    }
```
should work fine in the browser with this implementation